register.go

```diff
@@ -6,18 +6,18 @@ package fiber
 
 // Register defines all router handle interface generate by Route().
 type Register interface {
-	All(handlers ...Handler) Register
-	Get(handlers ...Handler) Register
-	Head(handlers ...Handler) Register
-	Post(handlers ...Handler) Register
-	Put(handlers ...Handler) Register
-	Delete(handlers ...Handler) Register
-	Connect(handlers ...Handler) Register
-	Options(handlers ...Handler) Register
-	Trace(handlers ...Handler) Register
-	Patch(handlers ...Handler) Register
-
-	Add(method string, handlers ...Handler) Register
+	All(handler Handler, middleware ...Handler) Register
+	Get(handler Handler, middleware ...Handler) Register
+	Head(handler Handler, middleware ...Handler) Register
+	Post(handler Handler, middleware ...Handler) Register
+	Put(handler Handler, middleware ...Handler) Register
+	Delete(handler Handler, middleware ...Handler) Register
+	Connect(handler Handler, middleware ...Handler) Register
+	Options(handler Handler, middleware ...Handler) Register
+	Trace(handler Handler, middleware ...Handler) Register
+	Patch(handler Handler, middleware ...Handler) Register
+
+	Add(methods []string, handler Handler, middleware ...Handler) Register
```

Comment:
we can allow simple strings and lists of strings here with generics ?

Reply:
I don't think generics is allowed in interface.

Reply:
One possibility is:

```
Add(methods interface{}, handler Handler, middleware ...Handler) Register calls
add(methods []string, handler Handler, middleware ...Handler) Register
```